x-pack/agent/pkg/agent/transpiler/ast.go

```diff
@@ -110,6 +115,20 @@ func (d *Dict) Hash() []byte {
 	return h.Sum(nil)
 }
 
+// AddKey add a key value to an existing dictionary.
+func (d *Dict) AddKey(k *Key) {
+	d.value = append(d.value, k)
```

Comment:
what about dups? should we check for overwrites?

Reply:
good point I think we should overwrite.